app/assets/javascripts/admin/order_cycles/controllers/order_cycle_exchanges_controller.js.coffee

```diff
@@ -51,9 +58,8 @@ angular.module('admin.orderCycles')
 
       incoming = true if $scope.view == 'incoming'
       params = { exchange_id: exchange.id, enterprise_id: exchange.enterprise_id, order_cycle_id: $scope.order_cycle.id, incoming: incoming, page: page}
-      ExchangeProduct.index params, (products, num_of_pages, num_of_products) ->
+      ExchangeProduct.index params, (products, num_of_pages) ->
         enterprise.num_of_pages = num_of_pages
-        enterprise.num_of_products = num_of_products
```

Comment:
instead of an extra method above "exchangeLoadedVariants" I'd implement the logic here in loadExchangeProducts.
instead of enterprise.num_of_products = num_of_products
we can do:

initialize enterprise.loaded_variants = 0 before the call to  ExchangeProduct.index
`enterprise.loaded_variants = 0`

and then inside:
```
angular.forEach products, (product) ->
        enterprise.loaded_variants += product.variants.length
```

Reply:
I can't tell if there's any real benefit in one over the other? Do you think it would be more efficient to increment the count as the variants are loaded?

Reply:
I think it would be slightly faster and less complex with one less method. but I dont think it's a massive difference, that's why I approve the PR as is. it's your call :-)
my first comment in 2020 happy new decade!

Reply:
Hm, maybe there is not much in it but I imagine Luis' version to be more efficient. The variants are counted once after they have been loaded. In the current version, they are calculated every time `exchangeLoadedVariants` is called. I don't know how often Angular executes that.

And since we are already talking about preferred styles, I have another version here: :wink: 
```coffee
enterprise.loaded_variants = products.reduce((sum, product) => sum + product.variants.length, 0))
```

Reply:
:-)